skill-tree /src/main/java/com/RDS/skilltree/Skill/SkillsServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.RDS.skilltree.Skill;
+
+import com.RDS.skilltree.User.UserModel;
+import com.RDS.skilltree.User.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@Slf4j
+public class SkillsServiceImpl implements SkillsService{
+    private final SkillRepository skillRepository;
+    private final UserRepository userRepository;
+    public SkillsServiceImpl(SkillRepository skillRepository,
+                             UserRepository userRepository){
+        this.userRepository = userRepository;
+        this.skillRepository = skillRepository;
+    }
```

Comment:
Nit: I think we should consider using @RequiredArgsConstructor to reduce boiler plate

Reply:
I think all args is same as the RequiredArgsConstructor here @Ajeyakrishna-k